android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/dao/UploadDAO.java

```diff
@@ -0,0 +1,64 @@
+package org.sagebionetworks.bridge.android.manager.dao;
+
+import android.content.Context;
+
+import com.google.common.collect.Sets;
+
+import org.sagebionetworks.bridge.android.manager.UploadManager;
+import org.sagebionetworks.bridge.rest.model.UploadSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+/**
+ * Created by jyliu on 3/10/2017.
+ */
+
+public class UploadDAO extends SharedPreferencesJsonDAO {
+    private static final Logger logger = LoggerFactory.getLogger(UploadDAO.class);
+
+    private static final String PREFERENCES_FILE  = "uploads";
```

Comment:
Storage Access is in RS, android-sdk has no RS dependency. You are on the right track that researchstack-sdk should implement StorageAccess or otherwise hook in the DAOs.